vavr/src/test/java/org/jdbi/v3/vavr/TestVavrValueArgumentFactory.java

```diff
@@ -17,64 +17,77 @@
 import java.util.Optional;
 
 import io.vavr.Lazy;
-import io.vavr.NotImplementedError;
 import io.vavr.control.Either;
 import io.vavr.control.Option;
 import io.vavr.control.Try;
 import io.vavr.control.Validation;
 import org.jdbi.v3.core.argument.Argument;
+import org.jdbi.v3.core.argument.Arguments;
 import org.jdbi.v3.core.config.ConfigRegistry;
 import org.jdbi.v3.core.generic.GenericType;
 import org.junit.Before;
+import org.junit.Rule;
 import org.junit.Test;
+import org.mockito.Mock;
+import org.mockito.junit.MockitoJUnit;
+import org.mockito.junit.MockitoRule;
 
 import static org.assertj.core.api.Assertions.assertThat;
 import static org.assertj.core.api.Assertions.assertThatThrownBy;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.when;
 
 public class TestVavrValueArgumentFactory {
+    private static final Type TRY_INTEGER = new GenericType<Try<Integer>>() {}.getType();
+    private static final Type OPTION_INTEGER = new GenericType<Option<Integer>>() {}.getType();
+    private static final Type LAZY_INTEGER = new GenericType<Lazy<Integer>>() {}.getType();
+    private static final Type LAZY_WILDCARD = new GenericType<Lazy<?>>() {}.getType();
+    private static final Type EITHER_STRING_INTEGER = new GenericType<Either<String, Integer>>() {}.getType();
+    private static final Type EITHER_WILDCARD = new GenericType<Either<?, ?>>() {}.getType();
+    private static final Type VALIDATION_STRING_INT = new GenericType<Validation<String, Integer>>() {}.getType();
 
-    private static final Argument MOCK_ARGUMENT = ((position, statement, ctx) -> new NotImplementedError());
+    @Rule
+    public MockitoRule mockito = MockitoJUnit.rule();
 
-    private VavrValueArgumentFactory unit;
+    @Mock
+    private ConfigRegistry configRegistry;
```

Comment:
Is there a particular reason this is written as a mock rather than ensuring we can e.g. actually insert null values into a h2 test table?

Reply:
The test was built on a fake Argument. I just replaced it with a proper way to fake the ConfigRegistry, Felt changing the "bigger picture" of the test was out of scope.

Reply:
And tbh this still rings in my mind

> This PR would have been a slam dunk if it just had the relevant fix at hand...

Reply:
I don't think that's an entirely fair comparison given that this test is introducing `mockito` and two new fixtures.  And part of the reason I was hoping for an actual test is that mocks often make for weak testing -- in this case, notably, we're just checking `assertThat(arg).isNotEmpty();` rather than making sure that an actual typed null gets the right sql type when it goes to the db vendor.  So the mocks are giving us a bit of a false sense of security here.

Reply:
So you wanna turn this simple bugfix (and collateral encapsulation to fix a gross leak) PR into one that also revises the relevant existing tests? Just to confirm

Reply:
In fact, I think the mocking is trivially unnecessary.  I'll push a commit showing what I mean momentarily.

Reply:
Cmon man, you're being really unfair.  Just because I want to remove mocking doesn't mean it's going to become gross.  Look at the change I made, don't you agree it's an improvement?

Reply:
? The "gross leak" referred to the non-`private` private method overridden for unit testing, which is the only collateral change I thought would be ok to include in a purely bugfix pr.

Reply:
whoops, I misread, sorry.  But hopefully you agree that removing the mocking makes the code nicer.